### PR TITLE
remove useless ident() functions in const tests

### PR DESCRIPTION
**Pull request by RalfJung**
and replace the useful ones by black_box (with a comment)

r? @oli-obk

### CODE REVIEW

File: src/test/run-pass/consts/const-ptr-nonnull.rs [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 const CASTED: NonNull<u32> = NonNull::cast(NonNull::<i32>::dangling());
 
-fn ident<T>(ident: T) -> T {
-    ident
-}
-
 pub fn main() {
-    assert_eq!(DANGLING, ident(NonNull::dangling()));
-    assert_eq!(CASTED, ident(NonNull::dangling()));
+    assert_eq!(DANGLING, b(NonNull::dangling()));
```
Comment (oli-obk) [inline src/test/run-pass/consts/const-ptr-nonnull.rs]: hmm... does this actually have the desired effect? I think we need to do `b(NonNull::dangling as fn() -> NonNull<u32>)()` in order to guarantee that rustc won't const prop the value by itself.

Comment (RalfJung) [reply src/test/run-pass/consts/const-ptr-nonnull.rs]: I don't think it is reasonable to fight against compiler optimizations like that here. I mean, it'll still get const-prop'ed by LLVM...

Comment (RalfJung) [reply src/test/run-pass/consts/const-ptr-nonnull.rs]: IOW, this is not the const-prop test suite. Then anyway we'd have to do three-way comparison to check both the result with and without const-prop.

Comment (oli-obk) [reply src/test/run-pass/consts/const-ptr-nonnull.rs]: const prop by LLVM is fine. The problem is that this is testing our const evaluator, and if the rhs is also computed by the const evaluator, then the comparison is moot.

I know it's not the const-prop test suite, it should not be testing const prop, it should be testing const eval against runtime. But with sufficiently smart const prop we'll get the rhs const propagated (at least the part inside the `b(..)`, const prop won't touch `black_box`).

Comment (RalfJung) [reply src/test/run-pass/consts/const-ptr-nonnull.rs]: The comparison is still relevant as it is the const evaluator accessed through very different code paths. We have (at least) three ways the computation might go here and this only compares two of them, I see no good reason why comparing these two would be better than comparing the other two. The const-prop test suite will make sure they are the same.

But as you wish, I changed it. Lucky enough this does not need a type annotation.

**Comment by oli-obk** [comment]
actually

@bors r-

you black-boxed the zst, which is a NOP. The cast to function pointers was on purpose

**Comment by RalfJung** [comment]
Now in extra-paranoid mode.

**Comment by oli-obk** [comment]
Thanks!

@bors r+

**Pull request merged**